src/AddTodoForm.js

```diff
@@ -1,18 +1,48 @@
 import React from "react";
 
 export default function AddTodoForm(props) {
+
+  // const {onAddTodo, todoTitle} = props; // applying destructuring
+
+  const [todoTitle, setTodoTitle] = React.useState("");
+
+  const handleTitleChange = (event) => {
+    let newTodoTitle = event.target.value;
+    setTodoTitle(newTodoTitle);
+    // console.log("newTodoTitle = " + newTodoTitle);
+  };
+
+  /// ----------------------------- ///
+
   const handleAddTodo = (event) => {
+
     event.preventDefault();
-    let todoTitle = event.target.title.value;
-    // console.log("todoTitle = " + todoTitle);
-    props.onAddTodo(todoTitle);
+    // let todoTitle = event.target.title.value; // old
+    // props.onAddTodo(todoTitle); // old
+
+    props.onAddTodo({ title: todoTitle, id: Date.now() });
+   
+   // Inside handleAddTodo, remove the reset() method and replace it with logic to reset the todoTitle state to an empty String
+
+    event.target[0].value="" // is this correct logic? If some mentor could verify in his feedback. Thx
+ 
+    // console.log(event.target[0].value);
+
+    
+
   };
 
   return (
     <div>
       <form onSubmit={handleAddTodo}>
         <label htmlFor="todoTitle">Title </label>
-        <input type="text" id="todoTitle" name="title"></input>
+        <input
+          type="text"
+          id="todoTitle"
+          name="title"
+          value={props.todoTitle} // there is no such parameter for this component on app.js line # 25  so how come no error flag?
```

Comment:
I think this was just supposed to be todoTitle from the state you created above. Not from the props because none were passed to the AddTodoForm component.